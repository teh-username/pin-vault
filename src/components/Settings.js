import React from 'react';
import { ScrollView } from 'react-native';
import { Card, List, ListItem, Text } from 'react-native-elements';

const Settings = ({
  requirePasscode,
  currentPasscode,
  handleTogglePasscodeRequirement,
  handleSetPasscode,
}) => (
  <List>
    <ListItem
      switchButton
      hideChevron
      switched={requirePasscode}
      title="Require Passcode"
      onSwitch={() =>
        handleTogglePasscodeRequirement(currentPasscode, requirePasscode)
      }
    />
    <ListItem title="Set Passcode" onPress={handleSetPasscode} />
  </List>
);

export default Settings;
