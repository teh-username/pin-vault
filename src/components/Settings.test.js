import React from 'react';

import Settings from './Settings';

describe('Settings component test', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper).toMatchSnapshot();
  });
});
