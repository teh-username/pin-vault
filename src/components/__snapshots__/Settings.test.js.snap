// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component test should render as expected 1`] = `
<List>
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={true}
    leftIconUnderlayColor="white"
    onSwitch={[Function]}
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitle="Enables passcode when opening the app"
    subtitleNumberOfLines={1}
    switchButton={true}
    textInputEditable={true}
    title="Passcode Protection"
    titleNumberOfLines={1}
    underlayColor="white"
  />
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="Set Passcode"
    titleNumberOfLines={1}
    underlayColor="white"
  />
</List>
`;
